lib/linked_data_fragments/backend_base.rb

```diff
@@ -1,6 +1,112 @@
 module LinkedDataFragments
+  ##
+  # A base class for RDF backends.
+  #
+  # Several abstract methods are defined, which implementations must provide 
+  # concrete versions of.
+  #
+  # @note this interface provides no method for clearing (deleting) individual
+  #   resources from the cache.
   class BackendBase
+    ##
+    # @!attribute [rw] cache_backend_url
+    #   @return [String, nil] a target url in string form; `nil` may be 
+    #     returned when the backend repository is not remote.
+    # @!attribute [rw] cache_backend_context
+    #   @return [String, nil] the context URI for the cache. When the backend 
+    #     supports quads, statements will be stored in this context. `nil` 
+    #     selects the default context.
+    #   @see https://www.w3.org/TR/rdf11-concepts/#section-dataset for 
+    #     information about named graphs ("contexts") in the RDF abstract syntax
+    #   @see RDF::Dataset for documentation covering named graph support in 
+    #     RDF.rb
+    #   @see RDF::Dataset#supports? for details about `:graph_name` support in 
+    #     Repositories
     attr_accessor :cache_backend_url, :cache_backend_context
+    
+    ##
+    # @abstract Add a resource to the backend. This method retrieves the 
+    #   resource from its URI.
+    # 
+    # @return [void]
+    # @raise [IOError, Net::HTTPError] when retrieving the resource fails with
+    #   a non-2xx HTTP status code.
+    # @see RDF::Graph#load  this is typical implementation
+    def add(uri)
+      raise NotImplementedError, 
+            "#{self.class} should implement `#empty?`, but does not."
```

Comment:
This comment should be something like the following (the current one seems to be an unchanged copy+paste): 

> raise NotImplementedError, 
>            "#{self.class} should implement `#add(uri)`, but does not."